native/packages/react-native-bpk-component-banner-alert/src/BpkBannerAlert.js

```diff
@@ -0,0 +1,298 @@
+    iconSource = tickIcon;
+    outerStyleFinal.push(styles.outerContainerSuccess);
+    iconStyle.push(styles.iconSuccess);
+  } else if (type === ALERT_TYPES.WARN) {
+    iconSource = informationIcon;
+    outerStyleFinal.push(styles.outerContainerWarn);
+    iconStyle.push(styles.iconWarn);
+  } else if (type === ALERT_TYPES.ERROR) {
+    iconSource = informationIcon;
+    outerStyleFinal.push(styles.outerContainerError);
+    iconStyle.push(styles.iconError);
+  }
+
+  if (dismissable) {
+    buttonIconSource = closeIcon;
+    buttonIconStyle.push(styles.closeIcon);
+  } else if (children !== null) {
+    if (expanded) {
+      buttonIconSource = chevronUpIcon;
+    } else {
+      buttonIconSource = chevronDownIcon;
+    }
+  }
+
+  const iconComponent = (
+    <Image
+      style={iconStyle}
+      source={iconSource}
+    />
+  );
+
+  const textComponent = (
+    <BpkText textStyle="sm" style={styles.text}>{message}</BpkText>
+  );
+
+  const actionComponent = (
+    <Image
+      style={buttonIconStyle}
+      source={buttonIconSource}
+    />
+  );
+
+  const closeButtonComponent = (
+    <TouchableHighlight
+      accessibilityComponentType="button"
+      onPress={onPress}
+      underlayColor={colorGray50}
+      accessibilityLabel={actionButtonLabel}
+      style={styles.closeButtonContainer}
+    >
+      {actionComponent}
+    </TouchableHighlight>
+    );
+
+  const contentPaddedStyle = [styles.bannerContainerPadded];
+  if (dismissable) {
+    contentPaddedStyle.push(styles.bannerContainerPaddedDismissable);
+  }
+
+  let banner = (<View
+    style={[contentPaddedStyle]}
+  >
+    {iconComponent}
+    {textComponent}
+    {expandable && actionComponent}
+  </View>
+  );
+
+  if (expandable) {
+    banner = (
+      <TouchableHighlight
+        accessibilityComponentType="button"
+        onPress={onPress}
+        underlayColor={colorGray50}
+        accessibilityLabel={actionButtonLabel}
+        style={styles.bannerContainer}
+      >
+        {banner}
+      </TouchableHighlight>
+    );
+  }
+
+  return (
+    <View style={outerStyleFinal} {...rest} >
+      <View style={styles.bannerContainer} >
+        {banner}
+        {dismissable && closeButtonComponent }
+      </View>
+      <BpkAnimateHeight expanded={expanded}>
+        <View style={expandedChildContainer}>{props.children}</View>
+      </BpkAnimateHeight>
+    </View>
+  );
+};
+
+BpkBannerAlert.propTypes = {
+  message: PropTypes.string.isRequired,
+  type: PropTypes.oneOf([
+    ALERT_TYPES.SUCCESS,
+    ALERT_TYPES.WARN,
+    ALERT_TYPES.ERROR,
+  ]).isRequired,
+  children: PropTypes.node,
+  dismissable: dismissablePropType,
+  expanded: PropTypes.bool,
+  onPress: PropTypes.func,
+  actionButtonLabel: PropTypes.oneOfType([
+    PropTypes.string,
+    PropTypes.func,
+  ]),
+  style: stylePropType,
+};
+
+BpkBannerAlert.defaultProps = {
+  children: null,
+  dismissable: false,
+  expanded: false,
+  onPress: () => null,
```

Comment:
It looks like this function is applied to all the `TouchableHighlight` so while closing, while expanding etc, is this intended behaviour?

Reply:
Expand / collapse etc should be left up to userland. The `onPress` function is there to allow them to respond to user interaction with the component

Reply:
my point is, there is one `onPress` and gets executed on both banner press and dismiss etc, I'm not sure you want the same function to run on both close and expand/collapse